### PR TITLE
[update to v0.8.0] autoware joy controller

### PR DESCRIPTION
**Pull request by TakaHoribe**
# TODO
 - [x] check usage of service client

### CODE REVIEW

File: control/autoware_joy_controller/include/autoware_joy_controller/joy_converter/ds4_joy_converter.hpp [hunk 1]
```diff
@@ -51,8 +51,8 @@ class DS4JoyConverter : public JoyConverterBase
 
   bool gate_mode() const {return Options();}
 
-  bool emergency() const {return !reverse() && PS();}
-  bool clear_emergency() const {return reverse() && PS();}
+  bool emergency_stop() const { return !reverse() && PS(); }
+  bool clear_emergency_stop() const { return reverse() && PS(); }
```
Comment (wep21) [inline control/autoware_joy_controller/include/autoware_joy_controller/joy_converter/ds4_joy_converter.hpp:56]: Are spaces unnecessary?

**Comment by wep21** [comment]
Build successful

File: control/autoware_joy_controller/include/autoware_joy_controller/joy_converter/ds4_joy_converter.hpp (hunk 1 above)
Comment (TakaHoribe) [reply control/autoware_joy_controller/include/autoware_joy_controller/joy_converter/ds4_joy_converter.hpp:56]: fixed

File: control/autoware_joy_controller/src/autoware_joy_controller/autoware_joy_controller_node.cpp [hunk 2]
```diff
@@ -308,21 +317,40 @@ void AutowareJoyControllerNode::publishGateMode()
...
+
+    if (!client_clear_emergency_stop_->wait_for_service(std::chrono::seconds(1))) {
+      RCLCPP_WARN(get_logger(), "failed to find clear_emergency_stop service");
+    } else {
+      auto request = std::make_shared<std_srvs::srv::Trigger::Request>();
+      auto result = client_clear_emergency_stop_->async_send_request(request);
+      if (
+        rclcpp::spin_until_future_complete(this->get_node_base_interface(), result) ==
+        rclcpp::FutureReturnCode::SUCCESS)
+      {
```
[the remark this answers is not in the document]
Comment (TakaHoribe) [reply control/autoware_joy_controller/src/autoware_joy_controller/autoware_joy_controller_node.cpp:340]: Sorry, I'll fix it.

[the remark this answers is not in the document]
Comment (TakaHoribe) [reply control/autoware_joy_controller/src/autoware_joy_controller/autoware_joy_controller_node.cpp]: fixed

**Comment by TakaHoribe** [comment]
Ready for review

File: control/autoware_joy_controller/launch/autoware_joy_controller.launch.xml [hunk 3]
```diff
@@ -24,7 +24,7 @@
     <remap from="output/shift" to="$(var output_shift)"/>
     <remap from="output/turn_signal" to="$(var output_turn_signal)"/>
     <remap from="output/gate_mode" to="$(var output_gate_mode)"/>
-    <remap from="output/emergency" to="$(var output_emergency)"/>
+    <remap from="output/emergency_stop" to="$(var emergency_stop)"/>
```
Comment (wep21) [inline control/autoware_joy_controller/launch/autoware_joy_controller.launch.xml:27]: var emergency_stop -> var output_emergency_stop?

Comment (TakaHoribe) [inline control/autoware_joy_controller/launch/autoware_joy_controller.launch.xml:27]:
```suggestion
    <remap from="output/emergency_stop" to="$(var output_emergency_stop)"/>
```

Comment (TakaHoribe) [reply control/autoware_joy_controller/launch/autoware_joy_controller.launch.xml]: thank you! 19af154

**Review by wep21 — APPROVED** [review]
LGTM

**Pull request merged**